feat: adicionar função de redefinir senha do Usuário

gabrielbdsm opened the pull request:
Isso inclui:
- Formulário para enviar email para redefirir senha
- Formulário para nova senha
- Modelo de redefinir senha no banco de dados
- Views para processar envio de email para redifinir senha
- Views para processar nova senha
- Validações da nova senha

 On branch feature/redefnirSenha
 Changes to be committed:
	new file:   barberease/usuarios/EmitEmail.py
	modified:   barberease/usuarios/forms.py
	modified:   barberease/usuarios/migrations/0001_initial.py
	deleted:    barberease/usuarios/migrations/0002_alter_usuario_sobrenome.py
	new file:   barberease/usuarios/migrations/0002_senharedefinicao.py
	new file:   barberease/usuarios/migrations/0003_alter_senharedefinicao_data_criacao.py
	modified:   barberease/usuarios/models.py
	modified:   barberease/usuarios/templates/registration/login.html
	new file:   barberease/usuarios/templates/usuario_new_password.html
	new file:   barberease/usuarios/templates/usuario_redefinir_senha.html
	modified:   barberease/usuarios/urls.py
	modified:   barberease/usuarios/views.py

romeuborges19 committed:
Merge pull request #18 from romeuborges19/develop

Develop

romeuborges19 committed:
Merge pull request #19 from romeuborges19/develop

fix: adicionando python-dotenv a requirements.txt

EnzoKaue committed:
Merge branch 'teste-nginx' of https://github.com/romeuborges19/projeto_barberease into teste-nginx

EnzoKaue committed:
fix settings.py

EnzoKaue committed:
mais fix no settings.py

romeuborges19 committed:
atualizando entrypoint.sh

romeuborges19 committed:
fix: atualizando settings.py

romeuborges19 committed:
fix: atualizando dockerfile

romeuborges19 committed:
fix: organizando arquivos

romeuborges19 committed:
fix: organizando links para styles nos templates

romeuborges19 committed:
fix: corrigindo problema com variavel global

romeuborges19 committed:
fix: testes para variavel should_redirect

EnzoKaue committed:
Merge pull request #26 from romeuborges19/teste-whitenoise

Adicionando whitenoise para deploy do website

ana-flav commented on barberease/usuarios/models.py:
```diff
@@ -17,4 +19,7 @@ def __str__(self):
     
 
- 
+class SenhaRedefinicao(models.Model):
```
Gabriel, cada classe criada no models é uma tabela criada no banco, no processo de redefinição de senha não é preciso criar mais uma tabela. para alterar uma senha voce tem que usar o models do usuario na alteração. Peço que modifique isso na seu codigo. Segue um link para ajudar [https://www.tutorialspoint.com/how-to-change-password-of-users-in-django](https://www.tutorialspoint.com/how-to-change-password-of-users-in-django). Tente usar esse metodo ou procure desde que a senha tenha hash e nao crie outra tabela no bd

ana-flav reviewed (changes requested):
alem dos comentários do models, antes de finalizar o seu trabalho lembre-se de dar o git pull origin develop para evitarmos muitos conflitos. de resto esta tudo ok

romeuborges19 commented on barberease/usuarios/forms.py:
```diff
@@ -47,3 +50,72 @@ def save(self, commit=True):
+        token = secrets.token_urlsafe(32)
+        
+        if SenhaRedefinicao.objects.filter(email=email).exists():
+            user = SenhaRedefinicao.objects.get(email=email)
```
Usar o model Usuario ao invés de criar um novo model, para evitar mal uso do banco de dados.

romeuborges19 commented on barberease/usuarios/forms.py:
```diff
@@ -47,3 +50,72 @@ def save(self, commit=True):
+
+    def process_password_reset(self):
+        email = self.cleaned_data.get('email')
+        token = secrets.token_urlsafe(32)
```
O Django tem funcionalidades de redefinição de senha que podem ser usadas para evitar a criação de um novo model no banco de dados. 

Seguem conteúdos sobre:
https://www.sjoerdlangkemper.nl/2016/04/07/djangos-reset-password-mechanism/
https://github.com/django/django/blob/main/django/contrib/auth/tokens.py#L8
https://docs.djangoproject.com/en/3.2/ref/settings/#password-reset-timeout

romeuborges19 reviewed (changes requested):
Realizar mudanças requisitadas.

gabrielbdsm committed:
feat: adicionar função de redefinir senha do Usuário

Isso inclui:

Formulário para enviar email para redefirir senha
Formulário para nova senha
Views para processar envio de email para redifinir senha
Views para processar nova senha
Validações da nova senha

Changes to be committed:
	modified:   barberease/settings.py
	new file:   usuarios/EmitEmail.py
	modified:   usuarios/forms.py
	modified:   usuarios/templates/registration/login.html
	new file:   usuarios/templates/usuario_new_password.html
	new file:   usuarios/templates/usuario_redefinir_senha.html
	new file:   usuarios/token.py
	modified:   usuarios/urls.py
	modified:   usuarios/views.py

romeuborges19 committed:
Merge branch 'develop' into feature/redefnirSenha

romeuborges19 reviewed (approved):
Top

gabrielbdsm committed:
style: Estilizei páginas de redefinição e nova senha

 Changes to be committed:
	modified:   usuarios/forms.py
	modified:   usuarios/static/usuarios/css/style.css
	modified:   usuarios/templates/usuario_new_password.html
	modified:   usuarios/templates/usuario_redefinir_senha.html
	modified:   usuarios/views.py

gabrielbdsm committed:
style: Estilizei páginas de redefinição e nova senha

gabrielbdsm committed:
style: Estilizei páginas de redefinição e nova senha

gabrielbdsm reviewed (commented):
alterações concluída

gabrielbdsm committed:
feat : Adicionado o botão de lembrou da senha
